client/my-sites/checkout/composite-checkout/components/contact-details-container.tsx

```diff
@@ -59,13 +59,21 @@ export default function ContactDetailsContainer( {
 		return null;
 	}
 
-	const { updateDomainContactFields, updateCountryCode, updatePostalCode, updateEmail } =
-		checkoutActions;
+	const {
+		updateDomainContactFields,
+		updateCountryCode,
+		updatePostalCode,
+		updateCity,
+		updateState,
+		updateEmail,
+	} = checkoutActions;
 	const contactDetails = prepareDomainContactDetails( contactInfo );
 	const contactDetailsErrors = prepareDomainContactDetailsErrors( contactInfo );
 	const onChangeContactInfo = ( newInfo: ManagedContactDetails ) => {
 		updateCountryCode( newInfo.countryCode?.value ?? '' );
 		updatePostalCode( newInfo.postalCode?.value ?? '' );
+		updateCity( newInfo.city?.value ?? '' );
+		updateState( newInfo.state?.value ?? '' );
```

Comment:
I think you need to add one of these (and all the other associated machinery) for the organization field also. When I type into it, nothing happens.

We might consider doing https://github.com/Automattic/wp-calypso/pull/72507 before this PR so that less code is needed and we would need only one call no matter how many pieces of data there are. I'll clean that PR up and put it back in review.